packages/dashboard/app/withdraw/WithdrawForm.tsx

```diff
@@ -0,0 +1,192 @@
+  const [selectedToken, setSelectedToken] = useState<TokenBalance | null>(null);
+  const [withdrawAmount, setWithdrawAmount] = useState<string>('');
+  const [withdrawAddress, setWithdrawAddress] = useState<string>('');
+  const [submitting, setSubmitting] = useState<boolean>(false);
+  const [statusMessage, setStatusMessage] = useState<string | null>(null);
+  const [statusType, setStatusType] = useState<StatusType>('info');
+  const [showHiddenTokens, setShowHiddenTokens] = useState<boolean>(false);
+  const hasInitializedRef = useRef<boolean>(false);
+  
+  const { showError } = useErrorPopup();
+  
+  // Ensure that we only fetch the balances once
+  useEffect(() => {
+    if (!shouldRefreshBalances) {
+      return;
+    }
+    if (!hasInitializedRef.current) {
+      hasInitializedRef.current = true;
+      refreshBalances();
+    }
+  }, [shouldRefreshBalances]);
+
+  // Auto-select ETH token when balances are loaded, ETH is guaranteed to be in the balances array
+  useEffect(() => {
+    if (balances.length > 0) {
+      const ethToken = balances.find(token => token.symbol === 'ETH');
+      handleTokenSelect(ethToken!, setSelectedToken, setWithdrawAmount);
+    }
+  }, [balances]);
+
+  // Wrapper for refresh balances
+  const refreshBalances = () => {
+    handleRefreshBalances(
+      agentPKP?.ethAddress,
+      setLoading,
+      setBalances,
+      showStatus,
+      showError
+    );
+  };
+  
+  // Use the hidden tokens hook
+  const { hiddenTokens, hideToken, unhideToken, error: hiddenTokensError } = useHiddenTokens();
+
+  useEffect(() => {
+    if (hiddenTokensError) {
+      showStatus(hiddenTokensError, 'error');
+    }
+  }, [hiddenTokensError]);
+
+  // Helper function to set status message
+  const showStatus = (message: string, type: StatusType = 'info') => {
+    setStatusMessage(message);
+    setStatusType(type);
+  };
+
+  // Toggle hidden tokens visibility
+  const toggleHiddenTokens = () => {
+    setShowHiddenTokens(!showHiddenTokens);
+  };
+
+  // Instead of state, use useMemo
+  const filteredBalances = useMemo(() => {
+    return showHiddenTokens 
+      ? balances 
+      : balances.filter(token => !hiddenTokens.includes(token.address.toLowerCase()));
+  }, [showHiddenTokens, hiddenTokens, balances]);
+
+  // Wrapper for token selection
+  const onSelectToken = (token: TokenBalance) => {
+    handleTokenSelect(token, setSelectedToken, setWithdrawAmount);
+  };
+
+  // Wrapper for max amount
+  const onMaxAmount = () => {
+    handleMaxAmount(selectedToken, setWithdrawAmount, showStatus);
+  };
+
+  // Wrapper for toggle hide token
+  const onToggleHideToken = (address: string, e: React.MouseEvent) => {
+    handleToggleHideToken(
+      address, 
+      e, 
+      hiddenTokens, 
+      selectedToken, 
+      hideToken, 
+      unhideToken, 
+      setSelectedToken, 
+      setWithdrawAmount
+    );
+  };
+
+  // Wrapper for submit
+  const onSubmit = (e: React.FormEvent) => {
+    handleSubmit(
```

Comment:
Both these functions are passing the event to a function that is not in "component land" and should not have to care about handling events.

The call to `e.preventDefault()` or `e.stopPropagating()` should be here and not have the event passed to the functions